slick.core.js

```diff
@@ -675,10 +726,251 @@
+      return true;
+    });
+  }
+  
+  function isHidden(el) {
+    return el.offsetWidth === 0 && el.offsetHeight === 0;
+  }
+
+  function parents(el, selector) {
+    const parents = [];
+    const visible = selector == ":visible";
+    const hidden = selector == ":hidden";
+
+    while ((el = el.parentNode) && el !== document) {
+
+      if(hidden)
+      {
+        if(isHidden(el)) {
+          parents.push(el);
+        }
+      }
+      else if (visible)
+      {
+        if(!isHidden(el)) {
+          parents.push(el);
+        }
+      }
+      else if (!selector || el.matches(selector)) 
+      {
+        parents.push(el);
+      }
+    }
+    return parents;
+  }
+
+  function template(html, parent) {
+    const template = document.createElement('template');
+    template.innerHTML = html.trim();
+
+    const first = template.content.firstChild;
+    if(parent)
+    {
+      [].forEach.call(template.content.children, function (child) {
+        parent.appendChild(child);
+      });
+      return first;
+    }
+    return first;
+  }
+  
+  function toFloat(value) {
+    var x = parseFloat(value)
+    if (isNaN(x)) {
+      return 0;
+    }
+    return x;
+  }
+
+  function show(el, type) {
+
+    type = type ? type : "";
+    if(Array.isArray(el)) {
+      el.forEach(function (e) {
+        e.style.display = type;
+      })
+    }
+    else {
+      el.style.display = type;
+    }
+  }
+
+  function hide(el) {
+    if(Array.isArray(el)) {
+      el.forEach(function (e) {
+        e.style.display = "none";
+      });
+    }
+    else {
+      el.style.display = "none";
+    }
+  }
+
+  function slideUp(el, callback) {
+    if(window.jQuery !== undefined) {
+      window.jQuery(el).slideUp("fast", callback);
+      return;
+    }
+
+    hide(el);
+    callback();
+  }
+
+  function slideDown(el, callback) {
+    if(window.jQuery !== undefined) {
+      window.jQuery(el).slideDown("fast", callback);
+      return;
+    }
+
+    show(el);
+    callback();
+  }
+
+  // jQuery's extend
+  var getProto = Object.getPrototypeOf;
+  var class2type = {};
+  var toString = class2type.toString;
+  var hasOwn = class2type.hasOwnProperty;
+  var fnToString = hasOwn.toString;
+  var ObjectFunctionString = fnToString.call( Object );
+  function isFunction( obj ) {
+    return typeof obj === "function" && typeof obj.nodeType !== "number" &&
+      typeof obj.item !== "function";
+  };
+  function isPlainObject( obj ) {
+    var proto, Ctor;
+    if ( !obj || toString.call( obj ) !== "[object Object]" ) {
+      return false;
+    }
+
+    proto = getProto( obj );
+    if ( !proto ) {
+      return true;
+    }
+    Ctor = hasOwn.call( proto, "constructor" ) && proto.constructor;
+    return typeof Ctor === "function" && fnToString.call( Ctor ) === ObjectFunctionString;
+  }
+  function extend() {
+    var options, name, src, copy, copyIsArray, clone,
+      target = arguments[ 0 ],
+      i = 1,
+      length = arguments.length,
+      deep = true;
```

Comment:
the `deep` default property is wrong, it should be set to `false` since that would follow `jQuery.extend()` more closely. I found this to be a problem when in my SlickGrid libs column filters were not showing up when it should. Basically, the following code in SlickGrid
```js
var m = columns[i] = utils.extend({}, columnDefaults, columns[i]);
```
is supposed to be a shallow copy (work as a pointer), however because of this `deep = true`, it actually became a deep copy and is causing issues in my lib because my lib assume these column copies are shallow copy and not deep copy. After changing default to `deep = false`, that would also follow more closely the NodeJS implementation in external 3rd party lib [node.extend](https://github.com/dreamerslab/node.extend)

https://github.com/dreamerslab/node.extend/blob/62adee3327630c83f3d39395f7e275bfba281ecc/lib/extend.js#L47-L55

```js
module.exports = function extend() {
  var target = arguments[0] || {};
  var i = 1;
  var length = arguments.length;
  var deep = false;
  var options, name, src, copy, copyIsArray, clone;
  // ....
```